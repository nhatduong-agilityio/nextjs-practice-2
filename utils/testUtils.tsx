import { ChakraProvider } from '@chakra-ui/react'
import { AuthProvider } from '@store/providers/AuthProvider'
import { LoadingProvider } from '@store/providers/LoadingProvider'
import type { RenderOptions, RenderResult } from '@testing-library/react'
import { render } from '@testing-library/react'
import { customTheme } from '@themes/index'
import type { ReactElement } from 'react'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <ChakraProvider theme={customTheme}>
      <LoadingProvider>
        <AuthProvider>{children}</AuthProvider>
      </LoadingProvider>
    </ChakraProvider>
  )
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
