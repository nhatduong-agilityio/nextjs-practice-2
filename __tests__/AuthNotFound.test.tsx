// Utils
import { render } from '@utils/testUtils'

// App
import NotFoundPage from '../app/(auth)/not-found'

describe('NotFoundPage render', () => {
  it('Should NotFoundPage match Snapshot', () => {
    const { container } = render(<NotFoundPage />)

    expect(container).toMatchSnapshot()
  })
})
