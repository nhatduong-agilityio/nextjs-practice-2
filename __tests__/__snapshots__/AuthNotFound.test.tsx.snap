// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage render Should NotFoundPage match Snapshot 1`] = `
<div>
  <div
    class="css-1wdafdy"
  >
    <p
      class="chakra-text css-bixgnl"
    >
      404 - Page Not Found!
    </p>
    <a
      href="/"
    >
      <p
        class="chakra-text css-xdjxp1"
      >
        Back To Home
      </p>
    </a>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
